packages/@rnw-scripts/integrate-rn/src/upgradeDependencies.ts

```diff
@@ -148,7 +187,22 @@ async function upgradeReactNative(
     }
   }
 
-  await runCommand('yarn install');
+  const repoRoot = await findRepoRoot();
+  const rootPkg = await WritableNpmPackage.fromPath(repoRoot);
+  if (rootPkg) {
+    if (rootPkg.json.resolutions && rootPkg.json.resolutions['react-native']) {
+      await rootPkg.mergeProps({
+        resolutions: {
+          'react-native': bumpSemver(
+            rootPkg.json.resolutions['react-native'],
+            newReactNativeVersion,
+          ),
+        },
+      });
+    }
+  }
+
+  await runCommand('yarn install --no-script');
```

Comment:
Why are you adding the --no-script here? I don't think we want to stop running `yarn build` here?

Reply:
We run `yarn install` multiple times in this script. There is no need to run `yarn build` the very first time. It will be run later anyway. Technically, it is not clear to me what it means to recompile the script that we currently run.
I believe it is a bug.
Anyway, I removed this change from the PR since I reverted the decision to use `yarn check` based on discussing with @acoates-ms.